frontend/src/app/search/error.tsx

```diff
@@ -0,0 +1,106 @@
+"use client"; // Error components must be Client Components
+
+import {
+  PaginationInfo,
+  SearchAPIResponse,
+} from "src/types/search/searchResponseTypes";
+
+import PageSEO from "src/components/PageSEO";
+import SearchCallToAction from "src/components/search/SearchCallToAction";
+import { SearchFetcherProps } from "src/services/search/searchfetcher/SearchFetcher";
+import { SearchForm } from "src/app/search/SearchForm";
+import { useEffect } from "react";
+
+interface ErrorProps {
+  // Next's error boundary also includes a reset function as a prop for retries,
+  // but it was not needed as users can retry with new inputs in the normal page flow.
+  error: Error & { digest?: string };
+}
+
+export interface ParsedError {
+  message: string;
+  searchInputs: SearchFetcherProps;
+  status: number;
+  type: string;
+}
+
+export default function Error({ error }: ErrorProps) {
+  // The error message is passed as an object that's been stringified.
+  // Parse it here.
+  const parsedErrorData = JSON.parse(error.message) as ParsedError;
```

Comment:
The errors are thrown in server components. Next strips out all properties on the error object that are not `message` - so we have to pass error metadata as stringified in the `error.message` - then JSON.parse it here